src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1385,6 +1408,19 @@ OGRGeometryH QgsOgrProvider::ConvertGeometryIfNecessary( OGRGeometryH hGeom )
   return OGR_G_ForceTo( hGeom, layerGeomType, nullptr );
 }
 
+QString QgsOgrProvider::jsonStringValue( const QVariant &value, const QVariant::Type &type )
+{
+  switch ( type )
+  {
+    case QVariant::Map:
+      return QString::fromUtf8( QJsonDocument::fromVariant( value.toMap() ).toJson().data() );
```

Comment:
I think the following change will work for map and list and make the code simpler.

```suggestion
       return QString::fromUtf8( QJsonDocument::fromVariant( value ).toJson().constData() );
```

Reply:
I cannot confirm that. `QString::fromUtf8( QJsonDocument::fromVariant( value ).toJson().constData() );` returns me an empty string if the value has a map structure.

Reply:
That is surprising.
The method http://doc.qt.io/qt-5/qjsondocument.html#fromVariant is not overloaded and just takes a single parameter QVariant (which should be one of QVariantMap, QVariantHash, QVariantList or QStringList). Can you see an explanation for the behavior you observed (I'm a bit lost)?

Reply:
Hm... Regarding the documentation I confirm your statement. And after some more tests it apparently works. I don't know what I did wrong 5 days ago, that I noticed the behavior that it does not work.